src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/RecalcEngineTests.cs

```diff
@@ -835,6 +835,50 @@ public void TestInheritanceOfDelegationWarningsInUDFs()
             Assert.True(result.IsSuccess);
             func = engine.Functions.WithName("NonDelegatableUDF2").First() as UserDefinedFunction;
             Assert.True(func.HasDelegationWarning);
+        }
+
+        [Theory]
+        [InlineData(
+            "DelegatableUDF(Value: Number):Boolean = Value > 5;",
+            "DelegatableUDF2():MyDataSourceTableType = Filter(MyDataSource, DelegatableUDF(Value));",
```

Comment:
>DelegatableUDF [](http://example.com/codeflow?start=76&length=14)

The fact that we can delegate this is surprising to me... do we expand the UDFs when translating it?

Reply:
We need to support it at least for non-row scope scenarios. For row-scope scenarios, it gets complicated in translation which we need to design it well.

Using UDF which is non-row scope shouldn't make the parent function call non-delegable. This works today at least which is good.

Reply:
For example,
UDF2(Name:Text) = Name;

This UDF2 could be row-scoped or non-row scoped depending in the context it is used.

Filter(Accounts, Name = UDF2(Name)) --> Row scoped
Filter(Accounts, Name = UDF2("")) --> Non-Row scoped